app/components/CreateRouteHoldsColor.js

```diff
@@ -16,8 +18,15 @@ class CreateRouteHoldsColor extends Component {
       h.style.backgroundColor = color;
     });
   }
+  checkHoldColor = holdColor => {
+    for (let key in holdColorDictonary) {
+      return holdColorDictonary[holdColor];
```

Comment:
A return statement will break the for loop and return you out of the function, so this is always going to return after the first iteration of the for loop

Reply:
I just wanted your insight of why we don't want to break out of the for loop and return out of the function after the iteration? My understanding is that the state (routeModel.holdColor) is changing when selecting the option which will run function and break out of the loop when the **value[key]** is found. 

The function will re-run again a new iteration will begin? and breakout of the loop when a different option selected has been chosen.

Reply:
@nries1 Hey I actually don't think we need a for loop and could just write the function cleaner as:

  **checkHoldColor = holdColor => {
    return holdColorDictonary[holdColor]
  };**

Reply:
Ye, you don't need a for loop. You don't even need a function really. All you need to do is look up the routeModel.holdColor in the holdColorDictionary

Reply:
Thanks, sorry I was just overthinking this way too much.....

Reply:
haha. no problem. It's fine to use a function. Just don't need the for loop